File: agent/src/main/java/com.evgenysobko.diploma.agent/TracerTrampoline.java
package com.evgenysobko.diploma.agent;

public final class TracerTrampoline {
    private static TracerHook hook = null;

    public static void installHook(TracerHook newHook) {
        hook = newHook;
    }

    public static void enter(int methodId, Object[] args) {
        hook.enter(methodId, args);
    }

    public static void leave() {
        hook.leave();
    }
}

File: src/main/kotlin/com/evgenysobko/diploma/agent/TracerTrampoline.kt
package com.evgenysobko.diploma.agent

object TracerTrampoline {
    private var hook: TracerHook? = null

    fun installHook(newHook: TracerHook?) {
        hook = newHook
    }

    fun enter(methodId: Int, args: Array<Any?>?) = hook!!.enter(methodId, args)

    fun leave() = hook!!.leave()
}